### DISCUSSION TITLE
does map work in application/x-scittle?

### DISCUSSION THREAD

**Discussion by kangaroolab** [borkdude/scittle#94] [General]
if you tried following:
- go to https://codepen.io/Prestance/pen/PoOdZQw?editors=1111
- add (map println [1 2 3]) at the of the script

nothing happens in the console. I also observed similar in my code where map in application/x-scittle doesn't seem to get executed so I have to change to doseq.

any insight what is happening?

**Comment by borkdude** [comment]
@kangaroolab you need to realize this sequence. map is lazy. so wrapping this in a doall will likely make it work.

**Comment by kangaroolab** [comment]
thanks...it works with doall/dorun...but any reason why it doesn't get executed in application/x-scittle, but execute as expected with core.eval_string?

**Reply by borkdude** [reply]
Perhaps because eval_script_tags doesn't return anything so nothing gets printed, but with eval_string the result gets printed and a as a result of printing the lazy seq gets realized? Feel free to provide a repro, else I'd be guessing what you're seeing.

**Comment by kangaroolab** [comment]
if you copy/paste following at codepen.io, it doesn't print anything in console

```
<html>
  <head>
    <script src="https://cdn.jsdelivr.net/npm/scittle@0.6.20/dist/scittle.js" type="application/javascript"></script>
  </head>
  <body>
    <script type="application/x-scittle">
      (map println [1 2 3])
    </script>
  </body>
</html>
```
but if you use dorun (see below), it works

```
<html>
  <head>
    <script src="https://cdn.jsdelivr.net/npm/scittle@0.6.20/dist/scittle.js" type="application/javascript"></script>
  </head>
  <body>
    <script type="application/x-scittle">
      (dorun (map println [1 2 3]))
    </script>
  </body>
</html>
```

**Reply by borkdude** [reply]
Yes, this is expected, since `map` returns a lazy seq and lazy seqs aren't realized if you don't... realize them.

**Comment by kangaroolab** [comment]
got it...thanks
